src/VisualStudio/Core/Def/TaskList/ExternalErrorDiagnosticUpdateSourceFactory.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Notification;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.TaskList
+{
+    [ExportWorkspaceServiceFactory(typeof(ExternalErrorDiagnosticUpdateSource), ServiceLayer.Host)]
+    [Shared]
+    internal sealed class ExternalErrorDiagnosticUpdateSourceFactory : IWorkspaceServiceFactory
+    {
+        private readonly IDiagnosticAnalyzerService _diagnosticService;
+        private readonly IDiagnosticUpdateSourceRegistrationService _registrationService;
+        private readonly IGlobalOperationNotificationService _notificationService;
+        private readonly IAsynchronousOperationListenerProvider _listenerProvider;
+        private readonly IThreadingContext _threadingContext;
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public ExternalErrorDiagnosticUpdateSourceFactory(
```

Comment:
>ExternalErrorDiagnosticUpdateSourceFactory [](http://example.com/codeflow?start=15&length=42)

Can this use primary constructor?

Reply:
Probably yes, but this syntax did not exist at the time this PR was authored. Planning to leave this as a follow-up.